website/pages/blog/accounts.js-1.0-rc.mdx

```diff
@@ -0,0 +1,217 @@
+*   Upgraded `@apollo/client` to 3.8
+*   Upgraded `@graphql-codegen` to v5
+*   Upgraded `mongodb` to v6
+*   Upgraded `ioredis` to v5
+*   Upgraded `jsonwebtoken` to v9
+*   Upgraded `lodash` to 4.17
+*   Upgraded `pg` to 8.11
+*   Upgraded `request-ip` to 3.3
+*   Upgraded `oauth` to 0.10
+*   Upgraded `node-fetch` to 2.7
+*   Upgraded `express` to 4.18
+*   Upgraded `emittery` to 0.13
+*   Upgraded `@levminer/speakeasy` to 1.4
+*   Upgraded `@graphql-tools/delegate` to v10
+*   Upgraded `@graphql-tools/stitch` to v9
+*   Upgraded `@graphql-tools/wrap` to v10
+*   Upgraded `mongoose` to v8
+*   Upgraded `react` to 18.2
+*   Upgraded `jest` to 29
+*   Upgraded `typescript` to 5.3
+*   Upgraded `eslint` to v8
+*   Upgraded `prettier` to v3
+*   Upgraded `@apollo/server` to 4.9
+*   Upgraded `docusaurus` to v3
+
+## What's left before 1.0 stable
+
+OAuth authentication, while working, surely deserves some love. While REST endpoints for OAuth should be functional there are no mutations nor resolvers yet, meaning that you can't use it with the GraphQL transport. I'd also like to go through the OAuth code and write some examples. `@accounts/oauth-instagram` in particular still relies upon the deprecated `request` package and should be updated. Other popular OAuth providers like Facebook still need to be added (but PRs exist).
+
+Before 1.0 stable gets released I plan to at least get the existing providers in a better shape, together with examples and the relevant GraphQL schema.
+
+## Post 1.0
+
+I'd like to create a new `@accounts/phone` authentication service which lets you authenticate via SMS OTPs. That would be especially useful in react-native applications where you could automatically read the sms and automate the authentication process.
+
+Currently Accounts.js bundles CommonJS code. While CommonJS can be imported in both CommonJS and ESM applications that would rule out Deno/Bun support. For the same reason we cannot use a wrapper either: while that would allow to use ESM imports it wouldn't be real ESM and thus won't be compatible. The remaning alternatives are pure ESM and dual packages. While several library authors opted for the former because of [dual package hazard](https://nodejs.org/api/packages.html#packages_dual_package_hazard) concerns I weight the benefits differently. While dual package hazards are real the whole GraphQL echosystem relies on dual packages thus using `instanceof` is already an hazard. These authors suggest to use `async imports` to import their pure ESM libraries in commonjs projects, which exposes everyone to the very same hazard (not even considering the fact that this is only viable in async contexts). That basically goes against the reason why they decided to bundle pure esm in the first place. I personally think that it's still too early to target pure esm and dual package is the lesser evil so I'm leaning towards that for future releases, but I'm ready to change my mind if you provide me enough arguments.
```

Comment:
it reads slightly annoyed, though
```suggestion
Currently, Accounts.js bundles CommonJS code. While CommonJS can be imported in both CommonJS and ESM applications, that would rule out Deno/Bun support. For the same reason, we cannot use a wrapper either: while that would allow us to use ESM imports, it wouldn't be real ESM and thus won't be compatible. The remaining alternatives are pure ESM and dual packages. While several library authors opted for the former because of [dual package hazard](https://nodejs.org/api/packages.html#packages_dual_package_hazard) concerns, I weigh the benefits differently. While dual package hazards are real, the whole GraphQL ecosystem relies on dual packages; thus, using `instance of` is already a hazard. These authors suggest using `async imports` to import their pure ESM libraries in common projects, which exposes everyone to the same hazard (not even considering that this is only viable in async contexts). That goes against why they decided to bundle pure ESM in the first place. I think it's still too early to target pure ESM, and dual package is the lesser evil, so I'm leaning towards that for future releases, but I'm ready to change my mind if you provide enough arguments.
```